scipy/stats/tests/test_stats.py

```diff
@@ -4142,6 +4142,18 @@ def test_weighted_stouffer2(self):
                                      weights=np.array((1, 4, 9)))
         assert_approx_equal(p, 0.1464, significant=4)
 
+    def test_pearson(self):
+        Z, p = stats.combine_pvalues([.01, .2, .3], method='pearson')
```

Comment:
where do these test values come from? is it really true that the combined p-value is 0.97 if one of the three is 0.01 and the other two are below 0.3 ?

Reply:
If you are asking where did `[.01, .2, .3]` numbers come from, that was copy and paste from the other unit tests.

This is the full number calculation that got me to that number.

(scipy-dev) Lucass-MacBook:stats rlucas$ python
Python 3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 14:01:38) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> pvalues = np.ndarray([.01, .2, .3])
>>> pvalues = np.array([.01, 0.2, 0.3])
>>> np.log1p(-pvalues)
array([-0.01005034, -0.22314355, -0.35667494])
>>> -2 * np.sum(np.log1p(-pvalues))
1.179737662212887
>>> statistic = -2 * np.sum(np.log1p(-pvalues))
>>> from scipy.stats import chi2
>>> chi2.sf(statistic, 2 * len(pvalues))
0.9778736999143088

They mention this is the linked arxiv paper too. Look at Figure 1 on page 2 here: https://arxiv.org/pdf/1707.06897.pdf

they also mention: 
`Tippett’s and Fisher’s methods are clearly more sensitive to the smallest
p-value, while Pearson’s method is most sensitive to the largest p-value.`

If you bring the 0.3 value up to something smaller this will bring down the test statistic, e.g. 

>>> pvalues2 = np.array([.01, 0.2, 0.9])
>>> statistic2 = -2 * np.sum(np.log1p(-pvalues2))
>>> chi2.sf(statistic2, 2 * len(pvalues2))
0.5346686266635268

this behavior stems from taking the 1-pvalue inside the logarithms, @chrisb83 does this answer your question? (I'm not sure if I'm answering what you are asking)